core/src/test/java/com/backblaze/b2/json/B2JsonBoundedByteArrayOutputStreamTest.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright 2021, Backblaze Inc. All Rights Reserved.
+ * License https://www.backblaze.com/using_b2_code.html
+ */
+package com.backblaze.b2.json;
+
+import com.backblaze.b2.util.B2StringUtil;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import java.io.IOException;
+import java.util.Arrays;
+
+import static junit.framework.TestCase.assertEquals;
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertNull;
+
+/**
+ * Unit test for B2JsonByteArrayOutputStream
+ */
+public class B2JsonBoundedByteArrayOutputStreamTest {
+
+    @Rule
+    public ExpectedException thrown  = ExpectedException.none();
+
+    @Test
+    public void testInitialState() {
+        final B2JsonBoundedByteArrayOutputStream b2JsonByteArrayOutputStream = new B2JsonBoundedByteArrayOutputStream(128);
+        assertArrayEquals(new byte[0], b2JsonByteArrayOutputStream.toByteArray());
+        assertEquals(128, b2JsonByteArrayOutputStream.getMaxCapacity());
+        assertEquals(0, b2JsonByteArrayOutputStream.getSize());
+    }
+
+    @Test
+    public void testWriteOneByte() throws IOException {
+        final B2JsonBoundedByteArrayOutputStream b2JsonByteArrayOutputStream = new B2JsonBoundedByteArrayOutputStream(2048);
```

Comment:
I think this would fail if the max were not a power of two.  that would be worth testing, too.